### PR TITLE
[ML-22328] Update README

### PR DESCRIPTION
**Pull request by jerrylian-db**
This PR updates the run-notebook GitHub action readme with content on how to:
* Create an Azure Service Principal via the Azure Portal.
* Generate Azure Active Directory tokens via an API call.
* Add Git credentials to the principal of the REST API token.
* Warn users to not use this action against workspaces with IP access restrictions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,37 +51,42 @@ and [generate an API token](https://docs.databricks.com/dev-tools/api/latest/tok
 For security reasons, we recommend using a Databricks service principal AAD token.
 
 ### Create an Azure Service Principal
-You can:
+Here are two ways that you can create an Azure Service Princiapl. The first way is via the UI. You follow do so in the Azure Portal, see the [Azure Databricks documentation](https://docs.microsoft.com/en-us/azure/databricks/administration-guide/users-groups/service-principals#create-a-service-principal). Record the Application (client) Id, Directory (tenant) Id, and client secret values generated by the steps.
```
Comment (mohamad-arabi) [inline README.md:54]: `You follow do so`, please reword this

Comment (mohamad-arabi) [reply README.md:54]: Also might be good to actually have the varaible names: e.g.  Application Id(`appId`)

File: README.md [hunk 2]
```diff
@@ -320,6 +314,10 @@ jobs:
 ### Run notebooks in different Databricks Workspaces
 In this example, we supply the `databricks-host` and `databricks-token` inputs
 to each `databricks/run-notebook` step to trigger notebook execution against different workspaces.
+The tokens are read from the GitHub repository secrets, `DATABRICKS_STAGING_TOKEN` and `DATABRICKS_STAGING_TOKEN` and `DATABRICKS_PROD_TOKEN`.
```
Comment (mohamad-arabi) [inline README.md:317]:
```suggestion
The tokens are read from the GitHub repository secrets, `DATABRICKS_DEV_TOKEN` and `DATABRICKS_STAGING_TOKEN` and `DATABRICKS_PROD_TOKEN`.
```

File: README.md [hunk 3]
```diff
@@ -377,37 +358,20 @@ jobs:
...
-          # AWS or "n1-highmem-4" for GCP
+          # The cluster JSON below is for AWS workspaces. On Azure and GCP, set
+          # node_type_id to an appropriate node type, e.g. "Standard_D3_v2" for
+          # Azure or "n1-highmem-4" for GCP
           new-cluster-json: >
             {
               "num_workers": 1,
               "spark_version": "10.4.x-scala2.12",
-              "node_type_id": "Standard_D3_v2"
+              "node_type_id": "i3.xlarge"
```
Comment (mohamad-arabi) [inline README.md:374]: Why are you setting node_type_id to AWS specific? Should we set these back to Azure specific e.g. `Standard_D3_v2`

File: README.md (hunk 1 above)
Comment (jerrylian-db) [reply README.md:54]: There are no variable names in the UI process, thus I have no included them.

File: README.md (hunk 3 above)
Comment (jerrylian-db) [reply README.md:374]: So that users will not have to generate an AAD token three times.

Comment (jerrylian-db) [reply README.md:374]: The thing is, AAD tokens only need to be generated once to be used across three workspaces (unless the total job time is over an hour)

Comment (jerrylian-db) [reply README.md:374]: I was concerned that by demonstrating generating the AAD token three times, users would assume that you needed to generate one for each workspace.

**Comment by mohamad-arabi** [comment]
LGTM once grammar issue is resolved here: https://github.com/databricks/run-notebook/pull/13#discussion_r878463119

**Pull request merged**